src/include/miopen/temp_file.hpp

```diff
@@ -36,23 +36,26 @@ namespace miopen {
 class TempFile
 {
     public:
-    TempFile(const std::string& path_template);
+    TempFile(const std::string& path_infix);
 
-    TempFile(TempFile&& other) noexcept : name(std::move(other.name)), dir(std::move(other.dir)) {}
+    TempFile(TempFile&& other) noexcept
+        : path_infix(std::move(other.path_infix)), dir(std::move(other.dir))
+    {
+    }
 
     TempFile& operator=(TempFile&& other) noexcept
     {
-        name = std::move(other.name);
-        dir  = std::move(other.dir);
+        path_infix = std::move(other.path_infix);
+        dir        = std::move(other.dir);
```

Comment:
Why is a custom assignment and move constructor necessary as there is no custom destructor in this class? Can't this be written as:

```cpp
TempFile(TempFile&& other) noexcept = default;
TempFile& operator=(TempFile&& other) noexcept = default;
```

Reply:
@pfultz2 Hmm... The default should work in this case.

@DrizztDoUrden This first appeared at https://github.com/ROCmSoftwarePlatform/MIOpen/pull/280/files#r775664957. Any special considerations there?

Reply:
@pfultz2 Resolved

Reply:
No, nothing special, I think.